### PR TITLE
PRJ-321: Mac local connection warning

### PR DESCRIPTION
**Pull request by IKrukov-HORIS**
This PR adds Idea platform GotIt messages for status bar widget:
- Hello message: greets the user and attracts user attention to Projector widget (once after install, but not often than one time per day). Plugin updates do not trigger the message.

- Mac Local Connection warning: alerts the user about keyboard input issue when connected locally on Mac (once per local connection)

### CODE REVIEW

File: projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/ProjectorStatusWidget.kt [hunk 1]
```diff
@@ -144,18 +151,42 @@ class ProjectorStatusWidget(private val myStatusBar: StatusBar)
...
     event?.let {
-      clients = event.newValue as Int
-      SwingUtilities.invokeLater { update() }
+      when (event.propertyName) {
+        "clientsCount" -> {
+          clients = event.newValue as Int
+          SwingUtilities.invokeLater { update() }
+        }
+
+        "macLocalConnection" -> {
```
Comment (SerVB) [inline projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/ProjectorStatusWidget.kt:168]: Could we maybe introduce named constants for these strings?

**Review by SerVB — CHANGES_REQUESTED** [review]
Cool notification but it can be improved.

I was able to test the Hello notification, but the notification for Mac doesn't show for me. Also, when a local client is connected to my Mac, a click to Projector widget doesn't work (no window is opened). Looks like a separate problem (I experience it in v1.7.0 too) but can be related: https://youtrack.jetbrains.com/issue/PRJ-740.

File: projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ProjectorInstallStateKeeper.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+class ProjectorInstallStateKeeper {
+  val isFirstRun = checkFirstRun()
+
+  private fun checkFirstRun(): Boolean {
+    val res = getFirstRunMark() == null
+    if (res) {
+      setFirstRunMark()
+    }
+
+    return res
```
[the remark this answers is not in the document]
Comment (IKrukov-HORIS) [reply projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ProjectorInstallStateKeeper.kt]: Fixed, renamed to isFirstRun

File: projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/ProjectorStatusWidget.kt (hunk 1 above)
Comment (IKrukov-HORIS) [reply projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/ProjectorStatusWidget.kt:168]: You a right. I've encapsulated the server -> plugin communication in ServerMessages module

**Review by SerVB — COMMENTED** [review]
PRJ-740 is fixed for me but Mac notification still isn't shown.

**Comment by IKrukov-HORIS** [comment]
>I was able to test the Hello notification, but the notification for Mac doesn't show for me. Also, when a local client is connected to my Mac, a click to Projector widget doesn't work (no window is opened). Looks like a separate problem (I experience it in v1.7.0 too) but can be related: https://youtrack.jetbrains.com/issue/PRJ-740.

You are right - both problems are related to ip v6 addresses support. Now both should be resolved.

**Review by SerVB — APPROVED** [review]
Works for me now. Shows a notification for local and ssh port forwarding connections on Mac. Doesn't show when connecting via IP.

**Pull request merged**